### PR TITLE
feat(analytics): swap remainder events

### PR DESCRIPTION
**Pull request by banklesss**
Related to [YOMO-740](https://emurgo.atlassian.net/browse/YOMO-740)

[YOMO-740]: https://emurgo.atlassian.net/browse/YOMO-740?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/ConfirmTxScreen.tsx [hunk 1]
```diff
@@ -38,6 +45,24 @@ export const ConfirmTxScreen = () => {
       signTx: {useErrorBoundary: true},
       submitTx: {
         onSuccess: () => {
+          const sellTokenInfo = tokenInfos.filter((tokenInfo) => tokenInfo.id === createOrder.amounts.sell.tokenId)[0]
+          const buyTokenInfo = tokenInfos.filter((tokenInfo) => tokenInfo.id === createOrder.amounts.buy.tokenId)[0]
```
Comment (banklesss) [inline apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/ConfirmTxScreen.tsx]: Should the context include the tokenInfo data beforehand?

Comment (stackchain) [reply apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/ConfirmTxScreen.tsx]: replace by useTokenInfo(sell)/(buy), avoid index accessing.

File: apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/mapOrders.ts [hunk 2]
```diff
@@ -1,7 +1,11 @@
 import {isString} from '@yoroi/common'
 import {getPoolUrlByProvider} from '@yoroi/swap'
 import {Balance} from '@yoroi/types'
+import {BalanceAmount, BalanceCardanoMetadatas} from '@yoroi/types/lib/balance/token'
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/mapOrders.ts]:
```suggestion
import {Balance, Swap} from '@yoroi/types'
```

File: apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/mapOrders.ts [hunk 3]
```diff
@@ -1,7 +1,11 @@
...
 import {SwapCompletedOrder, SwapOpenOrder} from '@yoroi/types/lib/swap/order'
+import {SwapPoolProvider} from '@yoroi/types/lib/swap/pool'
+import {BalanceTokenInfo} from '@yoroi/types/src/balance/token'
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/mapOrders.ts]:
```suggestion
```

File: apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/mapOrders.ts [hunk 4]
```diff
@@ -10,15 +14,36 @@ import {Quantities} from '../../../../../yoroi-wallets/utils'
...
+  tokenPrice: string
+  tokenAmount: string
+  id: string
+  assetFromLabel: string
+  assetToLabel: string
+  date: string
+  txId: string
+  total: string
+  txLink: string
+  toTokenInfo: BalanceTokenInfo<BalanceCardanoMetadatas> | undefined
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/mapOrders.ts]:
```suggestion
  toTokenInfo: Balance.TokenInfo | undefined
```

File: apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/mapOrders.ts [hunk 5]
```diff
@@ -10,15 +14,36 @@ import {Quantities} from '../../../../../yoroi-wallets/utils'
...
+  provider: SwapPoolProvider | undefined
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/mapOrders.ts]:
```suggestion
  provider: Swap.PoolProvider | undefined
```

File: apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/mapOrders.ts [hunk 6]
```diff
@@ -10,15 +14,36 @@ import {Quantities} from '../../../../../yoroi-wallets/utils'
...
+  poolUrl: string | undefined
+  fromTokenInfo: BalanceTokenInfo<BalanceCardanoMetadatas> | undefined
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/mapOrders.ts]:
```suggestion
  fromTokenInfo: Balance.TokenInfo | undefined
```

File: apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/mapOrders.ts [hunk 7]
```diff
@@ -10,15 +14,36 @@ import {Quantities} from '../../../../../yoroi-wallets/utils'
...
+  fromTokenAmount: string
+  from: BalanceAmount
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/mapOrders.ts]:
```suggestion
  from: Balance.Amount
```

File: apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/mapOrders.ts [hunk 8]
```diff
@@ -10,15 +14,36 @@ import {Quantities} from '../../../../../yoroi-wallets/utils'
...
+  to: BalanceAmount
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/mapOrders.ts]:
```suggestion
  to: Balance.Amount
```

File: apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/mapOrders.ts [hunk 9]
```diff
@@ -10,15 +14,36 @@ import {Quantities} from '../../../../../yoroi-wallets/utils'
...
   orders: Array<SwapOpenOrder | SwapCompletedOrder>,
   tokenInfos: Balance.TokenInfo[],
   numberLocale: NumberLocale,
   transactionInfos: TransactionInfo[],
-) => {
+): Array<MappedOrder> => {
   if (orders.length === 0) return []
 
-  return orders.map((order) => {
+  return orders.map((order: SwapOpenOrder | SwapCompletedOrder) => {
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/mapOrders.ts]:
```suggestion
  return orders.map((order: Swap.OpenOrder | Swap.CompletedOrder) => {
```

**Review by michalszorad — APPROVED** [review]

File: apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/mapOrders.ts [hunk 10]
```diff
@@ -1,6 +1,6 @@
 import {isString} from '@yoroi/common'
 import {getPoolUrlByProvider} from '@yoroi/swap'
-import {Balance} from '@yoroi/types'
+import {Balance, Swap} from '@yoroi/types'
 import {SwapCompletedOrder, SwapOpenOrder} from '@yoroi/types/lib/swap/order'
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/mapOrders.ts]:
```suggestion
```

File: apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/mapOrders.ts [hunk 11]
```diff
@@ -10,15 +10,36 @@ import {Quantities} from '../../../../../yoroi-wallets/utils'
...
+  provider: Swap.PoolProvider | undefined
+  poolUrl: string | undefined
+  fromTokenInfo: Balance.TokenInfo | undefined
+  fromTokenAmount: string
+  from: Balance.Amount
+  to: Balance.Amount
+}
+
 export const mapOrders = (
   orders: Array<SwapOpenOrder | SwapCompletedOrder>,
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/mapOrders.ts]:
```suggestion
  orders: Array<Swap.OpenOrder | Swap.CompletedOrder>,
```

**Pull request merged**